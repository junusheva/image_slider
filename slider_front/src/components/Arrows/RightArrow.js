import React from 'react';
import './Arrows.css';

const RightArrow = props => (
    <div className="rightArrow" onClick={props.goToNextSlide}>
        <i className="fas fa-angle-right"/>
    </div>

);

export default RightArrow;