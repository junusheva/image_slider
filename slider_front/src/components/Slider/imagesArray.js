const imageUrls = [
    //just some images from google
    "https://www.digitalcare.org/wp-content/uploads/2016/11/Free-Desktop-Wallpaper-feature.jpg",
    "https://cdn.shopify.com/s/files/1/0901/0816/products/Beautifully_Stranded_Desktop_1_2048x.JPG?v=1500905812",
    "https://cdn.wonderfulengineering.com/wp-content/uploads/2016/01/Desktop-Wallpaper-26.jpg",
    "https://cdn.cultofmac.com/wp-content/uploads/2019/01/A30A73E0-4AC8-412B-9803-557D2310BDDE.1af271bf674941aabb417ca24b32d194-780x566.jpeg",
    "https://static1.squarespace.com/static/552b1764e4b03cff36ace4b2/5a46c25424a694278eb57e34/5a46c4e2c83025f8444ca9b8/1514682496976/2017_September17.jpg",
    "https://cdn.newsapi.com.au/image/v1/06e4a0f5221e045e88ba3c707e1e7949"
];

export default imageUrls;