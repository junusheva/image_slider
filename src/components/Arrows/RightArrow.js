import React from 'react';

const RightArrow = props => (
    <div className="RightArrow" onClick={props.goToNextSlide}>
        <i className="fas fa-angle-left"/>
    </div>

);

export default RightArrow;