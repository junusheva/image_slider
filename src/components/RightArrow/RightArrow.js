import React from 'react';

const RightArrow = props => (
    <div className="RightArrow">
        <i className="fas fa-angle-left"/>
    </div>

);

export default RightArrow;