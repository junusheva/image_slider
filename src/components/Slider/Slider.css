html, body, .app, .main {
    height: 100%;
    width: 100%;
    margin: 0;
    padding: 0;
}

.slider {
    position: relative;
    width: 100%;
    margin: -5px auto 0;
    height: 95vh;
    overflow: hidden;
    white-space: nowrap;
}

.imagesWrapper {
    position: relative;
    height: 100%;
    width: 100%;
}