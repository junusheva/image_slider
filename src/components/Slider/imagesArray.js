const imageUrls = [
    "https://www.digitalcare.org/wp-content/uploads/2016/11/Free-Desktop-Wallpaper-feature.jpg",
    "https://cdn.shopify.com/s/files/1/0901/0816/products/Beautifully_Stranded_Desktop_1_2048x.JPG?v=1500905812",
    "https://cdn.wonderfulengineering.com/wp-content/uploads/2016/01/Desktop-Wallpaper-26.jpg",
    "https://cdn.shopify.com/s/files/1/0901/0816/products/Beautifully_Stranded_Desktop_1_2048x.JPG?v=1500905812",
    "https://cdn.wonderfulengineering.com/wp-content/uploads/2016/01/Desktop-Wallpaper-26.jpg"
];

export default imageUrls;