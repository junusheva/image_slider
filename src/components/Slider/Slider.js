import React, {Component} from 'react';
import './Slider.css';
import Slide from "../Slide/Slide";
import LeftArrow from "../LeftArrow/LeftArrow";
import RightArrow from "../RightArrow/RightArrow";

class Slider extends Component {

    render() {
        return (
            <div className="Slider">
                <Slide/>

                <LeftArrow/>
                <RightArrow/>
            </div>
        );
    }
}



export default Slider;
