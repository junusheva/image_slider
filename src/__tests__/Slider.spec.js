import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import Slider from '../components/Slider/Slider';
import RightArrow from "../components/Arrows/RightArrow";

describe('<Slider />', () => {

    it('creates a slider in an initial state', () => {

        let wrapper = renderer.create(<Slider/>).toJSON();
        expect(wrapper).toMatchSnapshot();

        const rightArrow = shallow(<RightArrow/>);
        rightArrow.simulate('click');

        wrapper = renderer.create(<Slider/>).toJSON();
        expect(wrapper).toMatchSnapshot();

    });

});