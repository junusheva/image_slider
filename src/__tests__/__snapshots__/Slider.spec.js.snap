// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> creates a slider in an initial state 1`] = `
<div
  className="slider"
>
  <div
    className="imagesWrapper"
    style={
      Object {
        "transform": "translateX(0px)",
        "transition": "transform ease-out 0.45s",
      }
    }
  >
    <div
      className="slide"
      style={
        Object {
          "backgroundImage": "url(https://www.digitalcare.org/wp-content/uploads/2016/11/Free-Desktop-Wallpaper-feature.jpg)",
          "backgroundPosition": "50% 60%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="slide"
      style={
        Object {
          "backgroundImage": "url(https://cdn.shopify.com/s/files/1/0901/0816/products/Beautifully_Stranded_Desktop_1_2048x.JPG?v=1500905812)",
          "backgroundPosition": "50% 60%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="slide"
      style={
        Object {
          "backgroundImage": "url(https://cdn.wonderfulengineering.com/wp-content/uploads/2016/01/Desktop-Wallpaper-26.jpg)",
          "backgroundPosition": "50% 60%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="slide"
      style={
        Object {
          "backgroundImage": "url(https://cdn.cultofmac.com/wp-content/uploads/2019/01/A30A73E0-4AC8-412B-9803-557D2310BDDE.1af271bf674941aabb417ca24b32d194-780x566.jpeg)",
          "backgroundPosition": "50% 60%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="slide"
      style={
        Object {
          "backgroundImage": "url(https://static1.squarespace.com/static/552b1764e4b03cff36ace4b2/5a46c25424a694278eb57e34/5a46c4e2c83025f8444ca9b8/1514682496976/2017_September17.jpg)",
          "backgroundPosition": "50% 60%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="slide"
      style={
        Object {
          "backgroundImage": "url(https://cdn.newsapi.com.au/image/v1/06e4a0f5221e045e88ba3c707e1e7949)",
          "backgroundPosition": "50% 60%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
  </div>
  <div
    className="leftArrow"
    onClick={[Function]}
  >
    <i
      className="fas fa-angle-left"
    />
  </div>
  <div
    className="rightArrow"
    onClick={[Function]}
  >
    <i
      className="fas fa-angle-right"
    />
  </div>
  <div
    className="dotsWrapper"
  >
    <div
      className="clickedDot"
      onClick={[Function]}
    />
    <div
      className="inactiveDot"
      onClick={[Function]}
    />
    <div
      className="inactiveDot"
      onClick={[Function]}
    />
    <div
      className="inactiveDot"
      onClick={[Function]}
    />
    <div
      className="inactiveDot"
      onClick={[Function]}
    />
    <div
      className="inactiveDot"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`<Slider /> creates a slider in an initial state 2`] = `
<div
  className="slider"
>
  <div
    className="imagesWrapper"
    style={
      Object {
        "transform": "translateX(0px)",
        "transition": "transform ease-out 0.45s",
      }
    }
  >
    <div
      className="slide"
      style={
        Object {
          "backgroundImage": "url(https://www.digitalcare.org/wp-content/uploads/2016/11/Free-Desktop-Wallpaper-feature.jpg)",
          "backgroundPosition": "50% 60%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="slide"
      style={
        Object {
          "backgroundImage": "url(https://cdn.shopify.com/s/files/1/0901/0816/products/Beautifully_Stranded_Desktop_1_2048x.JPG?v=1500905812)",
          "backgroundPosition": "50% 60%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="slide"
      style={
        Object {
          "backgroundImage": "url(https://cdn.wonderfulengineering.com/wp-content/uploads/2016/01/Desktop-Wallpaper-26.jpg)",
          "backgroundPosition": "50% 60%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="slide"
      style={
        Object {
          "backgroundImage": "url(https://cdn.cultofmac.com/wp-content/uploads/2019/01/A30A73E0-4AC8-412B-9803-557D2310BDDE.1af271bf674941aabb417ca24b32d194-780x566.jpeg)",
          "backgroundPosition": "50% 60%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="slide"
      style={
        Object {
          "backgroundImage": "url(https://static1.squarespace.com/static/552b1764e4b03cff36ace4b2/5a46c25424a694278eb57e34/5a46c4e2c83025f8444ca9b8/1514682496976/2017_September17.jpg)",
          "backgroundPosition": "50% 60%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="slide"
      style={
        Object {
          "backgroundImage": "url(https://cdn.newsapi.com.au/image/v1/06e4a0f5221e045e88ba3c707e1e7949)",
          "backgroundPosition": "50% 60%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
  </div>
  <div
    className="leftArrow"
    onClick={[Function]}
  >
    <i
      className="fas fa-angle-left"
    />
  </div>
  <div
    className="rightArrow"
    onClick={[Function]}
  >
    <i
      className="fas fa-angle-right"
    />
  </div>
  <div
    className="dotsWrapper"
  >
    <div
      className="clickedDot"
      onClick={[Function]}
    />
    <div
      className="inactiveDot"
      onClick={[Function]}
    />
    <div
      className="inactiveDot"
      onClick={[Function]}
    />
    <div
      className="inactiveDot"
      onClick={[Function]}
    />
    <div
      className="inactiveDot"
      onClick={[Function]}
    />
    <div
      className="inactiveDot"
      onClick={[Function]}
    />
  </div>
</div>
`;
